pkg/util/instanceselector/instanceselector.go

```diff
@@ -344,7 +344,7 @@ func toInstanceData(data []CustomInstanceData, memoryRequirement, cpuRequirement
 // the t2.Unlimited option from AWS. For T2 instances, we try to
 // figure out what percentage of a CPU a user will likely use and
 // use that to compute t2.Unlimited cost.
-func (instSel *instanceSelector) getInstanceFromResources(rs api.ResourceSpec, instanceTypeGlob string) (string, bool) {
+func (instSel *instanceSelector) getInstanceFromResources(rs api.ResourceSpec, additionalFilter func(inst InstanceData) bool) (string, bool) {
```

Comment:
I like that change, but it caused this snippet of code to be duplicated:

```golang
func(inst InstanceData) bool {
	// do any additional filtering that is needed
		if instanceTypeGlob == "" {
	matches = filterInstanceData(matches, additionalFilter)
			return true
		}
		return glob.Glob(instanceTypeGlob, inst.InstanceType)
	}
```

I suggest you add and use a helper function like this to remove the duplication:

```golang
func GlobInstanceType(string) func(InstanceData) bool
```